File: App/torneio_app/lib/core/models/mensagem_torneio.dart
class MensagemTorneio {
  final String id;
  final String torneioId;
  final String titulo;
  final String corpo;
  final String criadoPor;
  final DateTime criadoEm;

  const MensagemTorneio({
    required this.id,
    required this.torneioId,
    required this.titulo,
    required this.corpo,
    required this.criadoPor,
    required this.criadoEm,
  });

  factory MensagemTorneio.fromJson(Map<String, dynamic> json) => MensagemTorneio(
        id: json['id'] as String,
        torneioId: json['torneioId'] as String,
        titulo: json['titulo'] as String,
        corpo: json['corpo'] as String,
        criadoPor: json['criadoPor'] as String,
        criadoEm: DateTime.tryParse(json['criadoEm'] as String? ?? '')?.toLocal() ?? DateTime.now(),
      );
}

File: Retaguarda/Torneio.Domain/Entities/MensagemTorneio.cs
namespace Torneio.Domain.Entities;

public class MensagemTorneio
{
    public Guid Id { get; private set; }
    public Guid TorneioId { get; private set; }
    public string Titulo { get; private set; } = null!;
    public string Corpo { get; private set; } = null!;
    public string CriadoPor { get; private set; } = null!;
    public DateTime CriadoEm { get; private set; }

    private MensagemTorneio() { }

    public static MensagemTorneio Criar(Guid torneioId, string titulo, string corpo, string criadoPor) => new()
    {
        Id = Guid.NewGuid(),
        TorneioId = torneioId,
        Titulo = titulo,
        Corpo = corpo,
        CriadoPor = criadoPor,
        CriadoEm = DateTime.UtcNow
    };
}
